### PR TITLE
Added string representation to classes

### PR DESCRIPTION
**Pull request by DevayaniShivankar**
Please check _handleoutlier.py and _categorial.py

### CODE REVIEW

File: preprocessy/handlenullvalues/_main.py
Comment (Parth576) [inline preprocessy/handlenullvalues/_main.py:40]: I don't think we should print the dataframe, it makes the output a bit difficult to understand.

image

File: preprocessy/scaling/_main.py
Comment (Parth576) [inline preprocessy/scaling/_main.py:42]: The class name is ```Scaler```

File: preprocessy/scaling/errors.py
Comment (Parth576) [inline preprocessy/scaling/errors.py:10]: I don't think a ```__repr__``` function is required for this class.

File: preprocessy/encoding/_categorical.py
Comment (Parth576) [inline preprocessy/encoding/_categorical.py:43]: Why is ```self.params``` not being used here for the params argument?
Please remove the train and test dataframe from here as well, I have mentioned the reason in another comment.

**Review by Saif807380 — CHANGES_REQUESTED** [review]
Apart from all the changes that @Parth576 suggested, I think you should not use `self.argument` in `__repr__`. Instead, use the default values of the arguments there. 

For example,

```python
def __repr__(self):
     NullValuesHandler(df=None, ...)

     # rather than NullValuesHandler(df=self.df, ...)
```

File: preprocessy/handlenullvalues/_main.py
Comment (DevayaniShivankar) [reply preprocessy/handlenullvalues/_main.py:40]: Okay, on it

File: preprocessy/handlenullvalues/errors.py
Comment (DevayaniShivankar) [reply preprocessy/handlenullvalues/errors.py:10]: Will remove it

File: preprocessy/encoding/_categorical.py
Comment (DevayaniShivankar) [reply preprocessy/encoding/_categorical.py:43]: actually, params is of the form, 
params={
cat_cols = []
ord_dict = {}
}

So I made a default empty dictionary for that.

File: preprocessy/scaling/errors.py
Comment (DevayaniShivankar) [reply preprocessy/scaling/errors.py:10]: Will get it removed

**Review by Saif807380 — APPROVED** [review]
Great Work! Let's wait for @Parth576 before we merge it.

File: preprocessy/outliers/_handleoutlier.py
Comment (Saif807380) [inline preprocessy/outliers/_handleoutlier.py:59]: Rather than making an empty dictionary, use the values of the corresponding attributes of `self` if present else `None`.

File: preprocessy/resampling/_kfold.py
Comment (Saif807380) [inline preprocessy/resampling/_kfold.py:68]: Update them with corresponding `self.param_name`.

File: preprocessy/resampling/_split.py
Comment (Saif807380) [inline preprocessy/resampling/_split.py:25]: You can use `self.train_size`, `self.test_size` and `self.random_state` along with `None` for the dataframes.

File: preprocessy/scaling/_main.py
Comment (Saif807380) [inline preprocessy/scaling/_main.py:42]: Replace defaults for all params with `self.param_name` except for the dataframe param.

**Review by Saif807380 — CHANGES_REQUESTED** [review]
Apologies for the miscommunication earlier. The purpose of `__repr__` is to print the state of the object, hence the default values need to be replaced with the corresponding object attributes. 

Also, you missed `_categorical.py` in `feature_selection` module. You can use `None` and `self.threshold` as the values.

File: preprocessy/outliers/_handleoutlier.py
Comment (Parth576) [inline preprocessy/outliers/_handleoutlier.py:59]: I guess you forgot to resolve this: https://github.com/preprocessy/preprocessy/pull/37#discussion_r565840028

**Pull request closed without merging**
